javascript/ql/src/semmle/javascript/security/dataflow/CleartextLoggingCustomizations.qll

```diff
@@ -176,17 +176,48 @@ module CleartextLogging {
 
     override string describe() { result = "process environment" }
 
-    override DataFlow::FlowLabel getLabel() {
-      result.isTaint() or
-      result instanceof PartiallySensitiveMap
-    }
+    override DataFlow::FlowLabel getLabel() { result.isTaint() }
+  }
+
+  /**
+   * Holds if the edge `pred` -> `succ` should be sanitized for clear-text logging of sensitive information.
+   */
+  predicate isSanitizerEdge(DataFlow::Node pred, DataFlow::Node succ) {
+    succ.(DataFlow::PropRead).getBase() = pred
   }
 
   /**
-   * A flow label describing a map that might contain sensitive information in some properties.
-   * Property reads on such maps where the property name is fixed is unlikely to leak sensitive information.
+   * Holds if the edge `src` -> `trg` is an additional taint-step for clear-text logging of sensitive information.
    */
-  class PartiallySensitiveMap extends DataFlow::FlowLabel {
-    PartiallySensitiveMap() { this = "PartiallySensitiveMap" }
+  predicate isAdditionalTaintStep(DataFlow::Node src, DataFlow::Node trg) {
+    // A taint propagating data flow edge through objects: a tainted write taints the entire object.
+    exists(DataFlow::PropWrite write |
+      write.getRhs() = src and
+      trg.(DataFlow::SourceNode).flowsTo(write.getBase())
+    )
```

Comment:
That step was introduced in this PR: https://github.com/github/codeql/pull/2220, and the performance worked out. 

And we don't step out of arbitary PropReads, those [are sanitized](https://github.com/github/codeql/pull/3661/files#diff-887b3fe2f729518454d1b59dafe40f22R186). 

> I think we should use a label similar to TaintedObjectWrapper from TaintedObject.qll to avoid this, that is, a label that represents an object that deeply contains secret data.

I did try something similar to that label in https://github.com/github/codeql/pull/2220, but after [some back-and-forth](https://github.com/github/codeql/pull/2220#issuecomment-552351372) we ended up with the current approach.

Reply:
Ah, sorry I completely missed that. LGTM then